diff --git a/CMake/resolve_dependency_modules/cudf.cmake b/CMake/resolve_dependency_modules/cudf.cmake
index a13cd4c71..d44d87452 100644
--- a/CMake/resolve_dependency_modules/cudf.cmake
+++ b/CMake/resolve_dependency_modules/cudf.cmake
@@ -72,15 +72,15 @@ block(SCOPE_FOR VARIABLES)
 
   FetchContent_Declare(
     rapids-cmake
-    URL ${VELOX_rapids_cmake_SOURCE_URL}
-    URL_HASH ${VELOX_rapids_cmake_BUILD_SHA256_CHECKSUM}
+    GIT_REPOSITORY https://github.com/rapidsai/rapids-cmake.git
+    GIT_TAG ${VELOX_rapids_cmake_COMMIT}
     UPDATE_DISCONNECTED 1
   )
 
   FetchContent_Declare(
     rmm
-    URL ${VELOX_rmm_SOURCE_URL}
-    URL_HASH ${VELOX_rmm_BUILD_SHA256_CHECKSUM}
+    GIT_REPOSITORY https://github.com/rapidsai/rmm.git
+    GIT_TAG ${VELOX_rmm_COMMIT}
     SOURCE_SUBDIR
     cpp
     UPDATE_DISCONNECTED 1
@@ -88,8 +88,8 @@ block(SCOPE_FOR VARIABLES)
 
   FetchContent_Declare(
     kvikio
-    URL ${VELOX_kvikio_SOURCE_URL}
-    URL_HASH ${VELOX_kvikio_BUILD_SHA256_CHECKSUM}
+    GIT_REPOSITORY https://github.com/rapidsai/kvikio.git
+    GIT_TAG ${VELOX_kvikio_COMMIT}
     SOURCE_SUBDIR
     cpp
     UPDATE_DISCONNECTED 1
@@ -97,8 +97,8 @@ block(SCOPE_FOR VARIABLES)
 
   FetchContent_Declare(
     cudf
-    URL ${VELOX_cudf_SOURCE_URL}
-    URL_HASH ${VELOX_cudf_BUILD_SHA256_CHECKSUM}
+    GIT_REPOSITORY https://github.com/rapidsai/cudf.git
+    GIT_TAG ${VELOX_cudf_COMMIT}
     SOURCE_SUBDIR
     cpp
     UPDATE_DISCONNECTED 1
